src/btreemap.rs

```diff
@@ -2191,35 +2191,42 @@ mod test {
     #[should_panic(expected = "max_key_size must be <= 4")]
     fn rejects_larger_key_sizes() {
         let mem = make_memory();
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem, 5, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem, 5, 3);
     }
 
     #[test]
     fn accepts_small_or_equal_key_sizes() {
         let mem = make_memory();
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
         // Smaller key size
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem.clone(), 3, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem.clone(), 3, 3);
         // Equal key size
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem, 4, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem, 4, 3);
     }
 
     #[test]
     #[should_panic(expected = "max_value_size must be <= 3")]
     fn rejects_larger_value_sizes() {
         let mem = make_memory();
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem, 4, 4);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem, 4, 4);
     }
 
     #[test]
     fn accepts_small_or_equal_value_sizes() {
         let mem = make_memory();
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
         // Smaller key size
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem.clone(), 4, 2);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem.clone(), 4, 2);
         // Equal key size
-        let _btree: BTreeMap<_, Vec<u8>, Vec<u8>> = BTreeMap::init_with_sizes(mem, 4, 3);
+        let _btree: BTreeMap<Vec<u8>, Vec<u8>, _> = BTreeMap::init_with_sizes(mem, 4, 3);
+    }
+
+    #[test]
+    fn supports_unbounded_storable_types() {
+        let mem = make_memory();
+        // `String` doesn't implement `BoundedStorable` but can still be used in the `BTreeMap`.
+        let _btree: BTreeMap<String, String, _> = BTreeMap::init_with_sizes(mem.clone(), 4, 3);
```

Comment:
We should address the issue of using strings as keys by adding a bounded string type, `Text<CAP>`. This will make interfaces across data structures more consistent and will communicate the intent more clearly.